classes/issue/IssueDAO.inc.php

```diff
@@ -174,6 +174,51 @@ function getPublishedIssuesByNumber($journalId, $volume = null, $number = null,
 	}
 
 	/**
+	 * Retrieve Issues by identification
+	 * @param $journalId int
+	 * @param $volume int
+	 * @param $number string
+	 * @param $year int
+	 * @param $titles array
+	 * @return DAOResultFactory
+	 */
+	function getIssuesByIdentification($journalId, $volume = null, $number = null, $year = null, $titles = array()) {
+		$params = array((int) $journalId);
+
+		$sqlVol = $sqlNum = $sqlYear = '';
+		if ($volume !== null) {
+			$sqlVol = ' AND i.volume = ?';
+			$params[] = (int) $volume;
+		}
+		if ($number !== null) {
+			$sqlNum = ' AND i.number = ?';
+			$params[] = $number;
+		}
+		if ($year !== null) {
+			$sqlYear = ' AND i.year = ?';
+			$params[] = (int) $year;
+		}
+
+		$i = 1;
+		$sqlTitleJoin = $sqlTitleWhere = '';
+		foreach ($titles as $title) {
+			$sqlTitleJoin .= ' LEFT JOIN issue_settings iss' .$i .' ON (i.issue_id = iss' .$i .'.issue_id)';
+			$sqlTitleWhere .= ' AND iss' .$i .'.setting_name = \'title\' AND iss' .$i .'.setting_value = ?';
+			$params[] = $title;
+			$i++;
+		}
+
+		$sql = 'SELECT i.* FROM issues i'
+			.$sqlTitleJoin .
+			' WHERE i.journal_id = ?'
+			.$sqlVol .$sqlNum .$sqlYear
```

Comment:
Purely stylistic, but I prefer to keep all the SQL together in one string concatenation -- see e.g. `UserGroupDAO::getContextUsersCount`. It makes the query in all its permutations more readable IMO (though it does mean you need to track the order of parameters as they're added to the array).